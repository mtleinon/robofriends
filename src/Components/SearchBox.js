import React from 'react';

export const SearchBox = ({searchChange}) =>
    <div className='pa2'>
    <input className='pa3 ba bg-light-yellow'
           type='search'
           placeholder='Search robots'
           onChange={searchChange}/>;
    </div>