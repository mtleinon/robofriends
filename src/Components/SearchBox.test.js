import React from 'react';
import { shallow, mount, render } from 'enzyme';
import { SearchBox } from './SearchBox';

it('expect to render SearchBox component', () => {

    expect(shallow(<SearchBox/>)).toMatchSnapshot();
});