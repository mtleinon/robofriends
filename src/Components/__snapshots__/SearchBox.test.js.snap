// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render SearchBox component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchBox />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          aria-label="search robots"
          className="pa3 ba bg-light-yellow"
          placeholder="Search robots"
          type="search"
        />,
        ";",
      ],
      "className": "pa2",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "aria-label": "search robots",
          "className": "pa3 ba bg-light-yellow",
          "onChange": undefined,
          "placeholder": "Search robots",
          "type": "search",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      ";",
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            aria-label="search robots"
            className="pa3 ba bg-light-yellow"
            placeholder="Search robots"
            type="search"
          />,
          ";",
        ],
        "className": "pa2",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "aria-label": "search robots",
            "className": "pa3 ba bg-light-yellow",
            "onChange": undefined,
            "placeholder": "Search robots",
            "type": "search",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        ";",
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
